### PR TITLE
#2229 Address unnecessary requests that are made in react hooks based…

### PR DESCRIPTION
**Pull request by wu87988622**
… API

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2229 )
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/controller/useFetchApi.js [hunk 1]
```diff
@@ -24,29 +24,41 @@ const useFetchApi = url => {
...
+      }
+    };
+    fetchData();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [refetchState, request]);
 
   return { data: response, isLoading, refetch };
```
Comment (konekoya) [inline ohara-manager/client/src/components/controller/useFetchApi.js]: Hmm, how about making the refetch like: `refetch: () => refetch(true)`

I didn't test this. But it should work okay

File: ohara-manager/client/src/components/pages/NodePage/NodeListPage.js [hunk 2]
```diff
@@ -134,15 +134,15 @@ const NodeListPage = () => {
           isOpen={isNewModalOpen}
           handleClose={handleModalClose}
           handleConfirm={() => {
-            refetch();
+            refetch(true);
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/NodePage/NodeListPage.js]: With the above change, we won't need to pass the `true` here

**Comment by konekoya** [comment]
@wu87988622 could you also take a look at the API tests? Since we're using tags now. The tests should also assert that `tags` field is included in the API response

**Comment by wu87988622** [comment]
> @wu87988622 could you also take a look at the API tests? Since we're using tags now. The tests should also assert that `tags` field is included in the API response

OK I will add task in api test

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by wu87988622** [comment]
retry manager-e2e

**Pull request merged**